doc/monitoring_containers/monitoring_containers.md

```diff
@@ -0,0 +1,278 @@
+the unhealthy containers in SONiC.
+
+# Definitions/Abbreviation
+| Abbreviation |         Description          |
+|--------------|------------------------------|
+| Config DB    | SONiC Configuration Database |
+| CLI          | Command Line Interface       |
+
+# 1 Feature Overview
+SONiC is a collection of various switch applications which are held in docker containers
+such as BGP container and SNMP container. Each application usually includes several processes which are 
+working together to provide and receive the services from other modules. As such, the health of
+critical processes in each docker container is imperative not only for the docker
+container working correctly but also for the intended functionalities of entire SONiC switch.
+
+## 1.1 Monitoring
+This feature is used to monitor the running status of critical processes and critical resource 
+usage such as CPU, memory and disk of each docker container.
+
+We used Monit system tool to detect whether a critical process is running or not and whether 
+the resource usage of a docker container is beyond the pre-defined threshold.
+
+## 1.2 Auto-Mitigating
+This feature demonstrated docker container can be automatically shut down and
+restarted if one of critical processes running in docker container exits unexpectedly. Restarting
+the entire docker container ensures that configuration is reloaded and all processes in 
+docker container get restarted, thus increasing the likelihood of entering a healthy state.
+
+We leveraged the mechanism of event listener in supervisord to auto-restart a docker container
+if one of its critical processes exited unexpectedly. We also added a configuration option to make this 
+auto-restart feature dynamically configurable. Specifically users can run CLI to configure this 
+feature residing in Config_DB as enabled/disabled status.
+
+## 1.3 Requirements
+
+### 1.3.1 Functional Requirements
+1. Monit must provide the ability to generate an alert when a critical process has not
+    been alive for 5 minutes.
+2. Monit must provide the ability to generate an alert when the resource usage of
+    a docker container is larger than the pre-defined threshold.
+3. The event listener in supervisord must receive the signal when a critical process in 
+    a docker container crashed or exited unexpectedly and then restart this docker 
+    container.
+4. CONFIG_DB can be configured to enable/disable this auto-restart feature for each docker
+    container.. 
+5. Users can access the status of auto-restart feature via the CLI utility
+    1. Users can see current auto-restart status for docker containers.
+    2. Users can configure auto-restart status for a specific docker container.
+
+### 1.3.2 Configuration and Management Requirements
+Via the init_cfg.json file, these container features are disabled by default.
+Configuration of the auto-restart feature can be done via:
```

Comment:
I think you shouldn't limit this to the auto-restart feature. E.g., `Configuration of these container feature can be done via:`